temporalio/lib/temporalio/internal/worker/workflow_worker.rb

```diff
@@ -29,6 +29,12 @@ def self.workflow_definitions(workflows)
             # Confirm name not in use
             raise ArgumentError, "Multiple workflows named #{defn.name || '<dynamic>'}" if hash.key?(defn.name)
 
+            # Enforce versioning behavior is set when versioning is on
+            if should_enforce_versioning_behavior &&
+               defn.versioning_behavior == VersioningBehavior::UNSPECIFIED && defn.dynamic_options_method.nil?
```

Comment:
```suggestion
               defn.versioning_behavior == VersioningBehavior::UNSPECIFIED && !defn.dynamic_options_method
```

Pedantic, but Ruby is safer than Python in these cases because falsy in Ruby is only nil and false (not 0 or empty string or any of that). But this is fine too.